### PR TITLE
Upgrade clang-tidy.

### PR DESCRIPTION
**Pull request by trivialfis**

### CODE REVIEW

**Comment by trivialfis** [comment]
@hcho3 Hi, could you please help taking a look into installing clang-tidy-10 on Jenkins?  https://apt.llvm.org/  it doesn't seem to be using a key server and I'm getting a ` gpg: no valid OpenPGP data found.` error for accepting key directly.

**Comment by hcho3** [comment]
@trivialfis Fixed. Is clang-tidy 10 significantly better than clang-tidy 7? We are about to require all contributors to use cutting edge LLVM to run formatting checks.

**Comment by trivialfis** [comment]
@hcho3  I'm just trying to use latest clang-tidy on CI so that we don't stuck with old CUDA.  Users can specify `tidy-version` in our linting script.

**Comment by trivialfis** [comment]
@hcho3 You are a magician ..  I searched the internet for a long time with no clue ..

**Comment by RAMitchell** [comment]
> so that we don't stuck with old CUDA

What exactly does this mean?

I like the idea of upgrading but @hcho3 has a point that it makes it harder for new contributors. I'm hoping new clang-tidy significantly improves integration with cuda.

**Comment by trivialfis** [comment]
@RAMitchell 

> What exactly does this mean?

Clang supports cuda version by version: https://github.com/llvm-mirror/clang/blob/86c4707c4b8e0cd8045548d1a7fd35e37d7ccf71/lib/Driver/ToolChains/Cuda.cpp#L35

Currently employed clang-tidy-7 doesn't support CUDA 10,  so I just get the latest one.

File: include/xgboost/data.h [hunk 1]
```diff
@@ -42,34 +42,34 @@ class MetaInfo {
   static constexpr uint64_t kNumField = 9;
 
   /*! \brief number of rows in the data */
-  uint64_t num_row_{0};
+  uint64_t num_row_{0};  // NOLINT
```
Comment (hcho3) [inline include/xgboost/data.h:45]: Why `NOLINT` here?

Comment (trivialfis) [reply include/xgboost/data.h:45]: Wrong naming.  Public member should not end with `_`.  But I don't want to refactor it in this PR as it's too massive.

Comment (hcho3) [reply include/xgboost/data.h:45]: Got it.

**Pull request merged**